keps/sig-instrumentation/2831-kubelet-tracing/README.md

```diff
@@ -0,0 +1,354 @@
+  To be determined.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+  Upgrades and rollbacks will be tested while feature-gate is experimental
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+  No
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+###### How can an operator determine if the feature is in use by workloads?
+
+  Operators are expected to have access to and/or control of the OpenTelemetry agent deployment and trace storage backend.
+  KubeletConfiguration will show the FeatureGate and TracingConfiguration.
+
+###### How can someone using this feature know that it is working for their instance?
+
+  The tracing backend will display the traces with a service "kubelet".
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+  99% expected spans are collected from kubelet
+  No indications in logs of failing to export spans
+  
+##### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [] Metrics
+  - Metric name: tbd
+  - Components exposing the metric: kubelet
+
+##### Are there any missing metrics that would be useful to have to improve observability 
+  To be determined.
+
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+###### Does this feature depend on any specific services running in the cluster?**
+
+  Yes.  In the current version of the proposal, users must run the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector)
+  as a daemonset and configure a backend trace visualization tool (jaeger, zipkin, etc).
+
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+###### Will enabling / using this feature result in any new API calls?
+
+  This will not add any additional API calls.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+  This will introduce an API type for the configuration.  This is only for
+  loading configuration, users cannot create these objects.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+  Not directly.  Cloud providers could choose to send traces to their managed
+  trace backends, but this requires them to set up a telemetry pipeline as
+  described above.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+  A trace context, which is ~32 bytes is added to traced objects.
```

Comment:
What API objects are trace contexts added to? Is this an annotation on pods/containers?

Reply:
removed, thanks